xcodeproj/internal/xcodeproj_runner.bzl

```diff
@@ -115,7 +115,13 @@ def _write_schemes_json(*, actions, name, schemes_json):
     output = actions.declare_file(
         "{}-custom_xcode_schemes.json".format(name),
     )
-    actions.write(output, schemes_json if schemes_json else "[]")
+
+    # A new line character is added to make this file a valid unix text file,
+    # without this unexpect behaviours may occur when interacting with other
+    # unix-compatible tools (e.g. updating fixtures in the repo).
+    if schemes_json and not schemes_json.endswith("\n"):
```

Comment:
I saw many call sites so I thought it would be safer to assume that anything can be set as input there, but double checking now and it's all the fixture json being passed, also at the end is just an encode call and that always generates it without the `\n` indeed

I'll simplify then and take @mattrobmattrob's suggestion from the other thread